src/components/Game/Result/Result.js

```diff
@@ -1,36 +1,118 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { Redirect } from 'react-router-dom';
 import { connect } from 'react-redux';
+import classNames from 'classnames';
 
-const ResultView = ({ words, answer }) => {
-    const isAnswerCorrect = () => words.includes(answer);
-
-    return (
-        <div>
-            {
-                (isAnswerCorrect())
-                    ? 'gratulacje, wygrałeś'
-                    : (
-                        `nie udało się, może następnym razem.
-                        poprawne słowa to:
-                        ${words
-                            .map((word) => word.toUpperCase())
-                            .join(', ')}`
-                    )
-            }
-        </div>
-    );
-};
+import classes from './Result.scss';
+import { submitAnswer } from '../../../store/actions/answer';
+
+class ResultView extends Component {
+    componentWillUnmount() {
+        const { dispatch } = this.props;
+        dispatch(submitAnswer(''));
+    }
+
+    isAnswerCorrect = () => {
+        const { words, answer } = this.props;
+        return words.includes(answer);
+    };
+
+    handlePlayAgainClick = () => {
+        const { history } = this.props;
+        history.replace('/game');
+    };
+
+    render() {
+        const { words, answer } = this.props;
+
+        let redirect = null;
+        if (!answer) {
+            redirect = <Redirect to="/game" />;
```

Comment:
`history.replace`?
can it happen in a lifecycle method and not in the render method?

Reply:
I don't know 🤷🏻‍♀️ My goal was to redirect when refreshing `Result` page, going back to it after hitting `zagraj ponownie` or when trying to reach it by typing url into the browser. So if this is not correct, help needed!
https://i.pinimg.com/originals/73/44/c6/7344c6eb343fb4c6ad415a579c35fd83.gif

Reply:
The idea of redirecting is very good!
I would just change the way it's implemented. Now it's based on `Redirect` component, I would recommend using `history.replace` in a life cycle method.
I see https://reacttraining.com/react-router/web/api/Redirect provides an example similar to what you've done… I'm not 100% sure about the proper use case of `Redirect` but I think those kind of checks/guards should be done outside render function to keep the render function responsible for rendering only. But I imagine it's possible there are more experienced or smarter developers who disagree.

Reply:
If they (reacttraining) do it, I can't see a reason why I shouldn't. Maybe with a slight change - placing `Redirect` outside `return` like here: https://reacttraining.com/react-router/web/example/auth-workflow

Reply:
Definitely the version you just linked is cleaner. You have my blessing! :)